### PR TITLE
feat: add `toBeObject`

### PR DESCRIPTION
**Pull request by coratgerl**
### What does this PR do?

Add `toBeObject` from https://github.com/oven-sh/bun/issues/1825

- [x] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

 I wrote automated tests

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I included a test for the new code, or an existing test covers it
- [x] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [x] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: test/js/bun/test/expect.test.js [hunk 1]
```diff
@@ -3283,6 +3283,15 @@ describe("expect()", () => {
     expect({}).not.toBeInteger();
   });
 
+  test("toBeObject()", () => {
```
Comment (Electroid) [inline test/js/bun/test/expect.test.js:3286]: Could we add some more test cases here? e.g. `class A {}`, `[]`, `new Set()`, etc

File: src/bun.js/test/expect.zig [hunk 2]
```diff
@@ -2673,6 +2673,33 @@ pub const Expect = struct {
         return .zero;
     }
 
+    pub fn toBeObject(this: *Expect, globalThis: *JSGlobalObject, callFrame: *CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalThis);
+
+        const thisValue = callFrame.this();
+        const value: JSValue = this.getValue(globalThis, thisValue, "toBeObject", "") orelse return .zero;
+
+        active_test_expectation_counter.actual += 1;
```
Comment (Electroid) [inline src/bun.js/test/expect.zig]:
```suggestion
        incrementExpectCallCounter();
```

**Review by Electroid — CHANGES_REQUESTED** [review]

File: test/js/bun/test/expect.test.js (hunk 1 above)
Comment (coratgerl) [reply test/js/bun/test/expect.test.js:3286]: Yes of course, it's added !

**Comment by Electroid** [comment]
Thanks for working on this! Looks good, running the test suite now, and will have 1 other person review.

File: src/bun.js/test/expect.zig [hunk 3]
```diff
@@ -2673,6 +2673,33 @@ pub const Expect = struct {
...
+        const not = this.flags.not;
+        const pass = value.isObject() != not;
+
+        if (pass) return thisValue;
+
+        var formatter = JSC.ZigConsoleClient.Formatter{ .globalThis = globalThis, .quote_strings = true };
+        const received = value.toFmt(globalThis, &formatter);
+
+        if (not) {
+            const fmt = comptime getSignature("toBeObject", "", true) ++ "\n\n" ++ "Received: <red>{any}<r>\n";
```
Comment (dylan-conway) [inline src/bun.js/test/expect.zig]: could you add in a line above `Received: ...` that says `Expected value to be an object\n\n`? Similar to the formatting in `toBeEmpty`

File: src/bun.js/test/expect.zig [hunk 4]
```diff
@@ -2673,6 +2673,33 @@ pub const Expect = struct {
...
+            globalThis.throwPretty(fmt, .{received});
+            return .zero;
+        }
+
+        const fmt = comptime getSignature("toBeObject", "", false) ++ "\n\n" ++ "Received: <red>{any}<r>\n";
```
Comment (dylan-conway) [inline src/bun.js/test/expect.zig]: same here

**Review by dylan-conway — APPROVED** [review]
Thank you

**Pull request merged**